### PR TITLE
New Feature: Old Shaped Categories

### PR DESCRIPTION
**Pull request by WildmonkGitHub**
Resolves # This will bring the nostalgic of Scratch 2.0, because there are so many options in SA to make Scratch 3 look like Scratch 2. I've added one more feature that changes the shape of categories from round to a thin square.
Screenshot 2023-03-26 135303

### Changes

This feature is pre-release.

### Reason for changes

I found this idea when I saw some tutorials that how to make Scratch 3 look like Scratch 2 using SA, so I notice that there is one thing missing, and I made that - "Old Shaped Categories".

### Tests

I have tested this, there is no issue.

### CODE REVIEW

File: addons/shape-categories/addon.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    "credits": [
+      {
+        "name": "Wildmonk",
+        "link": "https://scratch.mit.edu/users/Wildmonk-2"
+      }
+    ],
+    "enabledByDefault": false,
+    "dynamicDisable": true,
+    "dynamicEnable": true,
+    "versionAdded": "1.31.1",
```
Comment (DNin01) [inline addons/shape-categories/addon.json:14]: We usually add addons in major versions (e.g., 1.32.0) only.

File: addons/shape-categories/addon.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{
+    "name": "Old Shaped Categories",
```
Comment (DNin01) [inline addons/shape-categories/addon.json:2]: Minor detail, all of the other addons (with a few exceptions) don't start every word with an uppercase letter. They actually look like this:
```suggestion
    "name": "Old shaped categories",
```

**Review by DNin01 — COMMENTED** [review]
This seems to work best in combination with the two-column category menu (`addons/columns`), so maybe this would work better as a setting for that addon instead of a new addon, unless you can find a way to make it compatible with that addon _and_ the normal design.

Be sure to test the addon with related addons enabled _and_ all other addons such as two-column category menu disabled.

**Comment by WorldLanguages** [comment]
> This seems to work best in combination with the two-column category menu (`addons/columns`), so maybe this would work better as a setting for that addon instead of a new addon

I initially agree with this

**Comment by Samq64** [comment]
What about the category icons addon?

**Comment by WorldLanguages** [comment]
> What about the category icons addon?

We could make it so that this new addon/setting overrides that addon (overriding the other way wouldn't make much sense).

**Comment by WorldLanguages** [comment]
We could also consider renaming `block palette category icons` and providing a setting on it that lets the user choose between the icons and this "old shaped thin square".

File: addons/shape-categories/addon.json (hunk 1 above)
Comment (WildmonkGitHub) [reply addons/shape-categories/addon.json]: I have changed it.

File: addons/shape-categories/addon.json (hunk 2 above)
Comment (WildmonkGitHub) [reply addons/shape-categories/addon.json]: Done with changes

**Comment by WildmonkGitHub** [comment]
> This seems to work best in combination with the two-column category menu (`addons/columns`), so maybe this would work better as a setting for that addon instead of a new addon, unless you can find a way to make it compatible with that addon _and_ the normal design.
> 
> Be sure to test the addon with related addons enabled _and_ all other addons such as two-column category menu disabled.

I have tested it, but it looks like it would not look good when you turn on (`block palette category icons`) while using this feature.

**Comment by DNin01** [comment]
> it looks like it would not look good when you turn on (`block palette category icons`) while using this feature.

That should definitely be considered.

**Comment by Samq64** [comment]
> We could also consider renaming `block palette category icons` and providing a setting on it that lets the user choose between the icons and this "old shaped thin square".

I think that's he best solution.

**Comment by WildmonkGitHub** [comment]
> That should definitely be considered. 

So, this feature will be not accepted unless I just fix the block palette category icon bug while using this feature.

**Review by DNin01 — CHANGES_REQUESTED** [review]
> > That should definitely be considered.
> 
> So, this feature will be not accepted unless I just fix the block palette category icon bug while using this feature.

Probably. It seems like a good thing to support. We don't really want one addon to conflict with another.

We could probably just move the icon _next_ to the category's color instead of inside it, only when this addon is enabled, of course.

Also, all by itself, it looks like this:

Colored bars replace the circles indicating category colors

I would expect the bars to be placed further to the left. This is why it's good to consider how an addon looks and behaves without the need for other addons. If we don't want to do that, then we could also consider moving this feature to another addon that it works well with (in this case, Two-column category menu), as I mentioned as an option earlier.

So, those are the things we need to do before we wrap up.

**Comment by Secret-chest** [comment]
> > > That should definitely be considered.
> > 
> > So, this feature will be not accepted unless I just fix the block palette category icon bug while using this feature.
> 
> Probably. It seems like a good thing to support. We don't really want one addon to conflict with another.
> 
> We could probably just move the icon _next_ to the category's color instead of inside it, only when this addon is enabled, of course.
> 
> Also, all by itself, it looks like this:
> 
> Colored bars replace the circles indicating category colors
> 
> I would expect the bars to be placed further to the left. This is why it's good to consider how an addon looks and behaves without the need for other addons. If we don't want to do that, then we could also consider moving this feature to another addon that it works well with (in this case, Two-column category menu), as I mentioned as an option earlier.
> 
> So, those are the things we need to do before we wrap up.

How about making it more compact (colours always on the same line?)

**Comment by WorldLanguages** [comment]
@WildmonkGitHub Why was this closed?

**Comment by WorldLanguages** [comment]
🤔

**Comment by WildmonkGitHub** [comment]
> 🤔

I don't think that it is a nice idea. I better work on other thing that has no problems.

**Comment by WorldLanguages** [comment]
I think the feature idea is nice. I guess you mean the implementation had problems.

**Comment by WildmonkGitHub** [comment]
> I think the feature idea is nice. I guess you mean the implementation had problems.

Better I work on next, this UI is not looking good in my opinion.

**Pull request closed without merging**